### PR TITLE
Diff dot ns frames

### PR DESCRIPTION
**Pull request by shireenf-ibm**
issue #279

### CODE REVIEW

File: pkg/netpol/internal/common/netpol_commands_common.go
Comment (adisos) [inline pkg/netpol/internal/common/netpol_commands_common.go:1]: consider splitting to another internal pkg related to the dot output formatting.

**Review by adisos — CHANGES_REQUESTED** [review]

File: pkg/netpol/internal/common/netpol_commands_common.go
Comment (shireenf-ibm) [reply pkg/netpol/internal/common/netpol_commands_common.go:1]: done

**Review by adisos — COMMENTED** [review]
why do the ci tests fail now?

**Comment by shireenf-ibm** [comment]
> why do the ci tests fail now?

in this branch I made minor changes to dot output (peers node name ) in both connlist and diff.
I need to run test-update for the new dot output that was added by the "connlist doc update branch" to get the fixed ones for this branch.

File: docs/diff_output.md [hunk 1]
```diff
@@ -34,26 +34,35 @@ added,0.0.0.0-255.255.255.255,default/backend[Deployment],No Connections,TCP 909
...
-        "default/frontend[Deployment]" -> "default/backend[Deployment]" [label="TCP 9090,UDP 53 (old: TCP 9090)" color="magenta" fontcolor="magenta"]
+        "default/frontend[Deployment]" -> "default/backend[Deployment]" [label="TCP 9090,UDP 53 (dir1: TCP 9090)" color="magenta" fontcolor="magenta"]
 }
 ```
 
 `svg` graph from `dot` format output can be produced using `graphviz` as following:
 
 ```
-$ dot -Tsvg tests/netpol-diff-example-minimal/diff_output_from_netpol-analysis-example-minimal.dot -o tests/netpol-diff-example-minimal/diff_output_from_netpol-analysis-example-minimal.svg
+$ dot -Tsvg test_outputs/diff/diff_between_netpol-diff-example-minimal_and_netpol-analysis-example-minimal.dot -o test_outputs/diff/diff_between_netpol-diff-example-minimal_and_netpol-analysis-example-minimal.dot.svg
```
Comment (adisos) [inline docs/diff_output.md:60]:
```suggestion
$ dot -Tsvg test_outputs/diff/diff_between_netpol-diff-example-minimal_and_netpol-analysis-example-minimal.dot -O

```

File: pkg/netpol/internal/dotformatting/dot_output_formatting.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	}
+	mapNsToPeers[peerNs] = append(mapNsToPeers[peerNs], "\t"+peerLine) // adding tab to the subgraph's lines
+}
+
+// return the label of a cluster's peer to be displayed on its node in the dot graph
+func NodeClusterPeerLabel(name, kind string) string {
+	return name + "[" + kind + "]"
+}
+
+// AddNsGroups adds namespaces frames to dot graphs
```
Comment (adisos) [inline pkg/netpol/internal/dotformatting/dot_output_formatting.go:27]: how does it add? what is the input/output, and what it does with it?

Comment (shireenf-ibm) [reply pkg/netpol/internal/dotformatting/dot_output_formatting.go]: updated the comment

**Review by adisos — APPROVED** [review]

**Pull request merged**